### PR TITLE
mh_hold_mode.pl v1.01

### PR DESCRIPTION
**Pull request by mh-source**
Emulation of ircII per-window hold_mode

changes:
nicer (imho) mh_sbmore

### CODE REVIEW

**Comment by ailin-nemui** [comment]
just using /window scroll on/off is not enough?

**Comment by mh-source** [comment]
Not if you want it "ircII"-like. it works a bit differently with regards to when it should scroll or not. i originally made this script for someone who was not happy with how irssi did its scroll, there might be others out there who also prefer the ircII way

File: scripts/mh_hold_mode.pl [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+#
+# irssi signal handlers
+#
+##############################################################################
+
+sub signal_gui_key_pressed_first
+{
+	my ($keycode) = @_;
+
+	if ($keycode == $KEYCODE_ENTER)
```
Comment (ailin-nemui) [inline scripts/mh_hold_mode.pl]: does it work if you remove this check here... (see below)

File: scripts/mh_hold_mode.pl [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+
+Irssi::statusbar_item_register('mh_sbmore', '', 'statusbar_more');
+Irssi::statusbars_recreate_items();
+
+statusbar_more_redraw();
+Irssi::timeout_add(1000, 'statusbar_more_redraw', undef);
+
+Irssi::command('^REDRAW');
+
+Irssi::signal_add_first('gui key pressed', 'signal_gui_key_pressed_first');
```
Comment (ailin-nemui) [inline scripts/mh_hold_mode.pl]: ...and use the signal "key send_line" instead of "gui key pressed"?

Comment (mh-source) [inline scripts/mh_hold_mode.pl]: I have just started going over it again (old script ive had lying around sincew 2007), so it was only cleaned up a little - didnt wanna mess with the internals till im a bit more into the code again. it does work as it is now, but i can look into using "key send_line" in future versions

**Comment by ailin-nemui** [comment]
hard coded values for the enter key may have been broken in the git version of irssi so unless you can confirm that it works in a git build of irssi this is not acceptable

**Comment by mh-source** [comment]
That is more effort than i want to put into this script. i will close the pull-request and just keep it on my own github page.

**Comment by ailin-nemui** [comment]
remember this thread whenever your distribution is going to upgrade irssi in 4-5 years ;-)

**Comment by mh-source** [comment]
Yeah, i will. Also warned my one known user that it might break in an upcoming version :-)  sounds a bit odd that the keycode have changed though.

**Comment by ailin-nemui** [comment]
The key codes have changed to make room for one additional key to use in custom key bindings, ^J. which was previously doubling as Enter

**Comment by mh-source** [comment]
Just my luck then. you wouldnt happen to know the new keycode? though i guess if i change it then it wont work in current irssi. using "key send_line" is probably a the right way forward

**Comment by ailin-nemui** [comment]
yes, if possible, using keycode-agnostic code would be better

**Comment by ailin-nemui** [comment]
I just submitted a PR for some scripts which makes them accept 10 and 13 to continue working with newer irssi. It's not perfect but will do for those old scripts I guess (keypad-enter key is not supported in that way)

**Comment by mh-source** [comment]
I still plan on looking into "key send_line", but i can change hold_mode to accept both 10 and 13 until then and make a PR?

**Pull request closed without merging**